### DISCUSSION TITLE
JSONSchemaViewer colors palette override

### DISCUSSION THREAD

**Discussion by manuelscurti** [jy95/docusaurus-json-schema-plugin#325] [Q&A] [answered]
Hi there! thanks for this wonderful plugin :)

Any hint on how to customize the colors used by the JSONSchemaViewer React component and override the ones given by docusaurus?

thanks!

**Comment by jy95** [comment]
Hello @manuelscurti, thanks ;)

Good question. I didn't tried this yet - would be curious about it.
Some hints could be : 

- https://docusaurus.io/docs/styling-layout#styling-your-site-with-infima
- https://infima.dev/
- https://infima.dev/docs/utilities/colors

The plugin uses at least the following things from Docusaurus color palette :

- classes `alert alert--info` for the collapse component ( https://infima.dev/docs/components/alert#info )
- `--ifm-color-primary` for the text
- `--ifm-background-color` for the background
- `--ifm-menu-color-background-active` for hovering
- `--ifm-color-secondary-darkest` for `constant`
- `--ifm-color-danger` for `required`
- `--ifm-color-primary-darker` for `writeOnly`
- `--ifm-color-info-darker` for `readOnly`
- `--ifm-color-warning` for `deprecated`

Maybe start with the [Global styles](https://docusaurus.io/docs/styling-layout#global-styles) approach - the documentation website has a [custom.css](https://github.com/jy95/docusaurus-json-schema-plugin/blob/main/testsite/src/css/custom.css) you can play with it . I have a Netlify set up as it is quite convenient to see visual changes in MR

**Reply by manuelscurti** [reply]
thanks for your answer. However, my docusaurus site already has its own theme (using the custom.css approach you mentioned) but for a specific page I would like the JSONSchemaViewer to have its own background color that is different from the one coming from Docusaurus theme. don't know if I made myself clear 👀

**Reply by jy95** [reply, marked as answer]
@manuelscurti Thanks for the clarification, I thought it wasn't the case yet.
I will publish the 1.12.0 version where you can put whatever [`className`](https://legacy.reactjs.org/docs/faq-styling.html) (by default, it will be `json-schema-viewer` ).

![image](https://github.com/jy95/docusaurus-json-schema-plugin/assets/9306961/5e4e055f-ff00-4726-9c5b-f218258491ba)

With [CSS Specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity), you can override the colors to your liking. I guess something like this could work : 

```css
/* "jsv-styles.module.css" for example*/
.json-schema-viewer {
   :root {
      --ifm-color-primary: #25c2a0;
      --ifm-code-font-size: 95%;
   }
}
```

For the way to add it to your website, either include the css file (with `customCss`) or using a wrapper class could do the job.

Of course, I would appreciate an real world example that I can put in [🛠️ Customizations showcases
](https://jy95.github.io/docusaurus-json-schema-plugin/docs/category/%EF%B8%8F-customizations) - would be handy for the future😉

**Reply by manuelscurti** [reply]
awesome! thanks ❤️
